src/design-system/layout/header/hamburger/hamburger.component.tsx

```diff
@@ -0,0 +1,23 @@
+import styles from "./hamburger.module.scss";
+import cx from "classnames";
+
+interface HamburgerProps {
+  setIsNavOpen: any; // Do zmiany typ (nie Function xd)
```

Comment:
setIsNavOpen: () => void;

jak potrzebujesz jakiegoś parametru w środku to tak
setIsNavOpen: (a: string) => void;

ogólnie wyodrębnij te funkcje od otwierania/zamykania menu do osobnego hooka use-header.hook.tsx (przykład użycia masz w use-navigation.hook.tsx)

i do tego headera daj to,

import { useRouter } from "next/router";
import { useEffect } from "react";
import { useHeaderDispatch } from "../providers/header.context";
import { useMainNavigationDispatch } from "../components/main-navigation/main-navigation.context";

export function useHeader() {
  const headerDispatch = useHeaderDispatch();
  const navDispatch = useMainNavigationDispatch();
  const router = useRouter();

  useEffect(() => {
    const handleRouteChange = () => {
      headerDispatch({ type: "HIDE_MENU" });
    };

    router.events.on("routeChangeStart", handleRouteChange);

    return () => {
      router.events.off("routeChangeStart", handleRouteChange);
    };
  }, [router, headerDispatch]);

  const closeMenu = () => {
    headerDispatch({ type: "HIDE_MENU" });
  };

  const showMenu = () => {
    headerDispatch({ type: "SHOW_MENU" });
  };

  return {
    closeMenu,
    showMenu,
  };
}

załatwi to przy okazji case, że schowa nawigacje na mobilce po kliknięciu w link